File: services/block-e-chunking/fixtures/code/javascript/logger.js
/**
 * Structured logging utility
 */

class Logger {
  constructor(name, level = 'info') {
    this.name = name;
    this.level = this._parseLevel(level);
    this.levels = {
      debug: 0,
      info: 1,
      warn: 2,
      error: 3
    };
  }

  _parseLevel(level) {
    return this.levels[level.toLowerCase()] || 1;
  }

  _shouldLog(level) {
    return this.levels[level] >= this.level;
  }

  _format(level, message, context = {}) {
    const timestamp = new Date().toISOString();
    return JSON.stringify({
      timestamp,
      level: level.toUpperCase(),
      logger: this.name,
      message,
      ...context
    });
  }

  debug(message, context = {}) {
    if (this._shouldLog('debug')) {
      console.log(this._format('debug', message, context));
    }
  }

  info(message, context = {}) {
    if (this._shouldLog('info')) {
      console.log(this._format('info', message, context));
    }
  }

  warn(message, context = {}) {
    if (this._shouldLog('warn')) {
      console.warn(this._format('warn', message, context));
    }
  }

  error(message, context = {}) {
    if (this._shouldLog('error')) {
      console.error(this._format('error', message, context));
    }
  }

  static create(name, level = 'info') {
    return new Logger(name, level);
  }
}

module.exports = Logger;

File: services/block-e-chunking/fixtures/code/python/logger.py
import logging
from typing import Optional
from datetime import datetime
import sys

class StructuredLogger:
    """Structured logging with context support."""
    
    def __init__(self, name: str, level: int = logging.INFO):
        self.logger = logging.getLogger(name)
        self.logger.setLevel(level)
        
        handler = logging.StreamHandler(sys.stdout)
        handler.setLevel(level)
        
        formatter = logging.Formatter(
            '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
        )
        handler.setFormatter(formatter)
        
        self.logger.addHandler(handler)
    
    def info(self, message: str, **context):
        """Log an info message with context."""
        self.logger.info(f"{message} {context or ''}")
    
    def error(self, message: str, **context):
        """Log an error message with context."""
        self.logger.error(f"{message} {context or ''}")
    
    def warning(self, message: str, **context):
        """Log a warning message with context."""
        self.logger.warning(f"{message} {context or ''}")
    
    def debug(self, message: str, **context):
        """Log a debug message with context."""
        self.logger.debug(f"{message} {context or ''}")
